sos/policies/redhat.py

```diff
@@ -284,6 +287,60 @@ def check(cls):
                         return True
         return False
 
+    def get_upload_url(self):
+        if self.case_id and not self.commons['cmdlineopts'].upload_url:
+            # clear the default FTP user name so that users are prompted for an
+            # RHCP username (or we're using a value from sos.conf)
+            self.upload_user = None
+            # this string is only used in pre_work() prompts, so this does not
+            # need to be the actual RHCP url
+            return "Red Hat Customer Portal"
+        return self.commons['cmdlineopts'].upload_url or RH_FTP_HOST
+
+    def get_upload_user(self):
+        # if this is anything other than dropbox, annonymous won't work
+        if self.get_upload_url() != RH_FTP_HOST:
+            return self.commons['cmdlineopts'].upload_user
+        return self.upload_user
+
+    def upload_archive(self, archive, url, directory, user, password):
+
+        # If either are missing, we cannot upload to a case. Fall back to
+        # dropbox
+        if not self.case_id or not password:
```

Comment:
Maybe extend it also by "or not user"? To catch (well, rare) situations `--batch --case=123 --upload-pass mysecret` without username, when we shall also fallback to dropbox.